src/snowflake/snowpark/mock/snowflake_data_type.py

```diff
@@ -95,7 +95,10 @@ def __getitem__(self, item):
 
     def __setitem__(self, key, value):
         super().__setitem__(key, value)
-        self.sf_types[key] = value.sf_type
+        if isinstance(
```

Comment:
Are all values here expected to be ColumnEmulator? cc: @sfc-gh-yixie 
I ran into when this is just `pd.Series`.

Reply:
I think so, what's the case that you run into `pd.Series`?